app/models/nic/bmc.rb

```diff
@@ -1,7 +1,14 @@
 module Nic
   class BMC < Managed
     PROVIDERS = %w(IPMI)
+    before_validation :ensure_physical
     validates :provider, :presence => true, :inclusion => { :in => PROVIDERS }
+    validates :mac, :presence => true, :if => :managed?
+
+    def virtual
```

Comment:
I get that, can you raise a `Foreman::Exception` for `def virtual=(value)` on Nic::BMC when virtual=true? That would highlight that the caller is doing something wrong, either Foreman or a user through the API could be setting this.